scylla/src/transport/connection.rs

```diff
@@ -61,9 +61,21 @@ pub struct Connection {
 
 type ResponseHandler = oneshot::Sender<Result<TaskResponse, QueryError>>;
 
-struct Task {
-    serialized_request: SerializedRequest,
-    response_handler: ResponseHandler,
+enum Task {
+    // Allocate a stream id for a request
+    AllocStreamId {
+        stream_id_sender: oneshot::Sender<i16>,
+        response_handler: ResponseHandler,
+    },
+    // Send a request to the server
+    Request {
+        stream_id: i16,
+        serialized_request: SerializedRequest,
+    },
```

Comment:
I think those should be a single message. Consider a situation in which the future returned from `send_request` sends `AllocStreamId`, then is dropped while waiting for the stream ID and does not send a `Request` message. The stream ID will not be freed in this case.

Reply:
Hm, right, makes sense to reduce the back-and-forth anyway. And all it takes is to add the stream_id_sender to `Request`. Will do in v2